react-app/src/components/InputText/InputText.js

```diff
@@ -0,0 +1,54 @@
+import React, { Component } from 'react';
+import FontAwesomeIcon from '@fortawesome/react-fontawesome';
+import { faEye, faEyeSlash } from '@fortawesome/fontawesome-free-solid';
+
+import './InputText.css';
+
+export default class InputText extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      type: this.props.type,
+      togglePasswordVisibilityButtonIcon: faEye
+    };
+
+    this.togglePasswordVisibility = this.togglePasswordVisibility.bind(this);
```

Comment:
You could use an arrow function as a public class attribute to get "auto binding". That's generally the recommended way of writing functions that need access to the instance data (e.g. to the state/props)